src/battle/area_dig/6DE1F0.c

```diff
@@ -2,6 +2,42 @@
 
 #define NAMESPACE b_area_dig
 
-INCLUDE_ASM(s32, "battle/area_dig/6DE1F0", func_80218430_6DE1F0);
+ApiStatus func_80218430_6DE1F0(Evt* script, s32 isInitialCall) {
+    f32 y;
+    f32 x;
+    f32 float3;
+    f32 angle;
+
+    s32 destBytecode1;
+    s32 destBytecode2;
+    s32 destBytecode3;
+
+    // Seems to be necessary to get the stack allocations to line up.
+    s8 unused[0x40];
+
+    s32* readPos = script->ptrReadPos;
+
+    angle = (f64)evt_get_variable(script, *readPos++) / 10.0f;
+    angle = angle * TAU;
+    angle = angle / 360.0f;
+
+    destBytecode1 = *readPos++;
+    destBytecode2 = *readPos++;
+    destBytecode3 = *readPos++;
+
+    y = sin_rad(angle) * 150.0f + 0.0f;
```

Comment:
I think the names of y and x should be switched here

Reply:
I matched y/x to sin/cos, but yeah it seemed fishy given that it stored y then x later. Will do.

Reply:
good point. I'm not sure..